### PR TITLE
remove _8 version of build, more doxygen, also comments in src/CMakeLists.txt

### LINKED ISSUE #79: Remove the _8 version of this library from the build
In this library we are building the _4, _d, and _8 variants.

Does anyone actually use the _8? On other NCEPLIBS we have removed it. For example, the g2 library only builds _d and _4.

Can anyone shed any light on whether the _8 version of sp (and ip) are ever used? I would love to drop it!

@GeorgeGayno-NOAA @GeorgeVandenberghe-NOAA @AlexanderRichert-NOAA @climbfuji @kgerheiser @Hang-Lei-NOAA

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #31 
Fixes #79

More doxygen. Some comments in CMakeLists.txt. Also removed the _8 build of the library.

### CODE REVIEW

**Comment by edwardhartnett** [comment]
OK, another batch of changes. Also added some comments to the src/CMakeLists.txt file.

More changes coming in future PRs.

File: src/spgrady.f [hunk 1]
```diff
@@ -1,36 +1,41 @@
...
-C> @param Q        - REAL ((M+1)*((I+1)*M+2)) SCALAR FIELD
-C> @param QDY      - REAL ((M+1)*((I+1)*M+2)) MERID GRADIENT (TIMES COSLAT)
-C> @param QDYTOP   - REAL (2*(M+1)) MERID GRADIENT (TIMES COSLAT) OVER TOP
+C> Extra terms are computed over top of the spectral domain.
+C>
+C> Advantage is taken of the fact that EPS(L,L)=0
+C> in order to vectorize over the entire spectral domain.
+C>
+C> @param I spectral domain shape
+c> (0 for triangular, 1 for rhomboidal)
```
Comment (AlexanderRichert-NOAA) [inline src/spgrady.f:21]: Put these on same line?

File: src/splaplac.f [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
+C> Computes the laplacian or the inverse laplacian
+c> of a scalar field in spectral space.
+C>      
+C> Subprogram speps() should be called already.
+C>      
+C> The Laplacian of Q(L,N) is simply -N*(N+1)/A**2*Q(L,N)
+C>
+C> @param I SPECTRAL DOMAIN SHAPE
+C> (0 FOR TRIANGULAR, 1 FOR RHOMBOIDAL)
+C> @param M SPECTRAL TRUNCATION
```
Comment (AlexanderRichert-NOAA) [inline src/splaplac.f]: Capitalization; put () part on previous line?

File: src/splaplac.f [hunk 3]
```diff
@@ -1,29 +1,32 @@
...
+C> @param ENN1 N*(N+1)/A**2
+C> @param[out] Q if IDIR > 0, scalar field
+C> (Q(0,0) is not computed)
```
Comment (AlexanderRichert-NOAA) [inline src/splaplac.f:17]: consolidate lines

File: src/sptgpsv.f [hunk 4]
```diff
@@ -1,19 +1,28 @@
 C> @file
+C> @brief Transform spectral vector to polar stereo.
 C>
-C> Transform spectral vector to polar stereo.
-C> @author IREDELL @date 96-02-29
+C> ### Program History Log
+C> Date | Programmer | Comments
+C> -----|------------|---------
+C> 96-02-29 | IREDELL | Initial
+C> 1998-12-15 | IREDELL | openmp directives inserted
```
Comment (AlexanderRichert-NOAA) [inline src/sptgpsv.f]: punctuation

File: src/sptgps.f [hunk 5]
```diff
@@ -1,15 +1,24 @@
 C> @file
+C> @brief Transform spectral scalar to polar stereo.
 C>
-C> Transform spectral scalar to polar stereo.
-C> @author IREDELL @date 96-02-29
+C> ### Program History Log
+C> Date | Programmer | Comments
+C> -----|------------|---------
+C> 96-02-29 | IREDELL | Initial
+C> 1998-12-15 | IREDELL | openmp directives inserted
```
Comment (AlexanderRichert-NOAA) [inline src/sptgps.f]: punctuation

File: src/sptgpt.f [hunk 6]
```diff
@@ -1,47 +1,50 @@
...
+C> @brief Transform spectral scalar to station points.
 C>
-C> Transform spectral scalar to station points
-C> @author IREDELL @date 96-02-29
+C> ### Program History Log
+C> Date | Programmer | Comments
+C> -----|------------|---------
+C> 96-02-29 | IREDELL | Initial.
+C> 1998-12-15 | IREDELL | openmp directives inserted
+C> 2003-06-30 | IREDELL | use spfftpt()
```
Comment (AlexanderRichert-NOAA) [inline src/sptgpt.f]: caps/punctuation

File: src/sptgptd.f [hunk 7]
```diff
@@ -1,46 +1,50 @@
 C> @file
+C> @brief Transform spectral to station point gradients.
 C>
-C> Transform spectral to station point gradients
-C> @author IREDELL @date 96-02-29
+C> ### Program History Log
+C> Date | Programmer | Comments
+C> -----|------------|---------
+C> 96-02-29 | IREDELL | Initial.
+C> 1998-12-15 | IREDELL | openmp directives inserted
```
Comment (AlexanderRichert-NOAA) [inline src/sptgptd.f]: caps/punctuation

File: src/sptgptsd.f [hunk 8]
```diff
@@ -1,7 +1,12 @@
 C> @file
+C> @brief Transform spectral scalar to station points.
 C>
-C> Transform spectral scalar to station points
-C> @author IREDELL @date 96-02-29
+C> PROGRAM HISTORY LOG:
+C> -  96-02-29  IREDELL
+C> - 1998-12-15  IREDELL  OPENMP DIRECTIVES INSERTED
+C> - 1999-08-18  IREDELL  OPENMP DIRECTIVE TYPO FIXED
```
Comment (AlexanderRichert-NOAA) [inline src/sptgptsd.f]: table formatting/caps/punctuation

File: src/sptgptsd.f [hunk 9]
```diff
@@ -1,7 +1,12 @@
...
-C> Transform spectral scalar to station points
-C> @author IREDELL @date 96-02-29
+C> PROGRAM HISTORY LOG:
+C> -  96-02-29  IREDELL
+C> - 1998-12-15  IREDELL  OPENMP DIRECTIVES INSERTED
+C> - 1999-08-18  IREDELL  OPENMP DIRECTIVE TYPO FIXED 
+C>
+C> @author Iredell @date 96-02-29
 
 C> THIS SUBPROGRAM PERFORMS A SPHERICAL TRANSFORM
```
Comment (AlexanderRichert-NOAA) [inline src/sptgptsd.f]: capitalization (throughout)

File: src/sptgptvd.f [hunk 10]
```diff
@@ -1,66 +1,65 @@
 C> @file
+C> @brief Transform spectral vector to station points.
 C>
-C> Transform spectral vector to station points
-C> @author IREDELL @date 96-02-29
+C> 96-02-29 | IREDELL | Initial
+C> 1998-12-15 | IREDELL | openmp directives inserted
+C> 1999-08-18 | iredell | openmp directive typo fixed
```
Comment (AlexanderRichert-NOAA) [inline src/sptgptvd.f]: caps/punctuation

File: src/spgrady.f (hunk 1 above)
Comment (edwardhartnett) [reply src/spgrady.f:21]: Doxygen will do that. If there is not an empty comment line, it will treat the text as one paragraph in html.

**Review by AlexanderRichert-NOAA — APPROVED** [review]

File: src/splaplac.f (hunk 2 above)
Comment (edwardhartnett) [reply src/splaplac.f]: Fixes capitalization.

File: src/sptgpsv.f (hunk 4 above)
Comment (edwardhartnett) [reply src/sptgpsv.f]: Fixed

File: src/sptgps.f (hunk 5 above)
Comment (edwardhartnett) [reply src/sptgps.f]: Fixed.

File: src/sptgptd.f (hunk 7 above)
Comment (edwardhartnett) [reply src/sptgptd.f]: Fixed.

File: src/sptgptsd.f (hunk 9 above)
Comment (edwardhartnett) [reply src/sptgptsd.f]: Fixed.

File: src/sptgptvd.f (hunk 10 above)
Comment (edwardhartnett) [reply src/sptgptvd.f]: Fixed

File: src/sptgptsd.f (hunk 8 above)
Comment (AlexanderRichert-NOAA) [reply src/sptgptsd.f]: @edwardhartnett I don't see any changes to sptgptsd.f? Both in terms of table fmt and capitalization of the main doc block.

Comment (edwardhartnett) [reply src/sptgptsd.f]: Check again? Looks good to me on my branch, and there are no outstanding commits to be pushed...

**Pull request merged**